library/std/src/sys/windows/thread_local_key.rs

```diff
@@ -111,16 +111,13 @@ struct Node {
 }
 
 unsafe fn register_dtor(key: Key, dtor: Dtor) {
-    let mut node = Box::new(Node { key, dtor, next: ptr::null_mut() });
+    let mut node = ManuallyDrop::new(Box::new(Node { key, dtor, next: ptr::null_mut() }));
```

Comment:
Since we never drop this anyway, couldn't we also just `Box::into_raw` here and then `compare_exchange`ing the raw pointer directly? I guess that would mean `node.next = head` needs to be `(*node).next = head` and is now unsafe, so maybe the ManuallyDrop is better, but I am somewhat unhappy with the `&mut **node` (the previous `&mut *node` was not better imo). I know we can rely on the deref ops on boxes, but it still seems wrong to me to not be using `Box::into_raw` when this is technically what we want.

Reply:
> I guess that would mean node.next = head needs to be (*node).next = head and is now unsafe

I went with `into_raw` first and decided to use `ManuallyDrop` when I realized this.